plugins/importexport/native/NativeImportExportPlugin.inc.php

```diff
@@ -291,17 +291,28 @@ function executeCLI($scriptName, &$args) {
 					return;
 				}
 
-				$filter = 'native-xml=>monograph';
-				// is this monographs import:
+				$request = Application::get()->getRequest();
+				// Set global user
+				if (!$request->getUser()) {
+					Registry::set('user', $user);
+				}
+				// Set global context
+				if (!$request->getContext()) {
+					HookRegistry::register('Router::getRequestedContextPaths', function (string $hook, array $args) use ($press): bool {
+						$args[0] = [$press->getPath()];
+						return false;
+					});
+					$router = new PageRouter();
+					$router->setApplication(Application::get());
+					$request->setRouter($router);
```

Comment:
This was the least aggressive and "honest" way I could come up with to reset the main context... A less honest way, would be to update the private variables of the router, but I preferred to not follow this path.

Possible issues: code caching the router/context locally in a variable/closure/etc.